runtime/onert/core/include/util/ObjectManager.h

```diff
@@ -135,11 +141,33 @@ template <typename Index, typename Object> class ObjectManager
   }
 
 private:
-  Index generateIndex() { return Index{_index_count++}; }
+  Index generateIndex(Index index = Index{})
+  {
+    // _next_index is always maintained to be "the biggest index in the object map" + 1
+    if (index.valid())
+    {
+      if (_objects.find(index) == _objects.end())
+      {
+        // If the given index does not exist, update the next index and return the index
+        if (index.value() >= _next_index)
+          _next_index = index.value() + 1;
+        return index;
+      }
+      else
+      {
+        // If the given index exists already, return a non-valid index
+        return Index{};
+      }
+    }
+    else
+    {
+      return Index{_next_index++};
+    }
```

Comment:
@wateret Till now, we've assumed `_next_index` would not reach `UINT32_MAX` because it is always incremented by 1. Now, I've considered what will happen if user specify `UINT32_MAX - 1` as index. At the call,  `_next_index` will be `UINT32_MAX` (= invalid index) by line 153. Then, at the next call of `generateIndex()`, It will reach the lines 162-165 and return invalid index. Furthermore, if the caller does not check the return value, at the next calls, it will return 1, 2, 3, .... which is likely to be used by others.

Reply:
I have updated it. As the logic gets a little complicated, I separated the methods that used to get an index as an optional param. Thanks!